### ISSUE TITLE
Unsupported ELF relocations: R_ARM_THM_MOVW_ABS_NC, R_ARM_THM_MOVT_ABS

### ISSUE THREAD

**Issue by denys-volodarskyi** [Vector35/binaryninja-api#6215]
**Version and Platform (required):**
- Binary Ninja Version: 4.2.6455 Personal,  2c8da1e
- OS: windows
- OS Version: 11
- CPU Architecture: x86_64

**Bug Description:**
Unsupported ELF relocation: R_ARM_THM_MOVW_ABS_NC
Unsupported ELF relocation: R_ARM_THM_MOVT_ABS

How can I upload sample file privately?

**Comment by psifertex** [comment]
For a private upload send it as an attachment or a link to binaryninja@vector35.com. Thanks.

**Comment by psifertex** [comment]
Available internally at V35 as `story squash popular warning`

**Comment by galenbwill** [comment]
Should be fixed in builds >= 4.3.6806.

**Comment by denys-volodarskyi** [comment]
5.0.7252-dev Personal (45e0f4e1)

Relocations are parsed, but seems not applied. There must be real symbol instead of nullptr.

![Image](https://github.com/user-attachments/assets/f9ded86d-8b37-4f06-b7b7-b649fc7c4811)

**Comment by galenbwill** [comment]
I'm not sure if your new screenshot is from the same binary, but looking at the binary you submitted before, I see this:

![Image](https://github.com/user-attachments/assets/c2270fdb-15b4-4e36-a418-d499a9eb2417)

but looking at the relocations using `readelf`, I see this:
```
000001ac  00008402 R_ARM_ABS32            00008389   AAA@A@@@
```
so that relocation, for example, is not one of the `R_ARM_THM_MOV*_ABS*` relocations, it is an absolute relocation. But the offset 0x8389 would be outside the mapped sections of this binary. I remember looking at this before when I was fixing the issue.

If you have more insight into what's going wrong here, please share. But this is not a regression -- the 4.3.6806 version that closed this issue shows the same output.

**Comment by denys-volodarskyi** [comment]
This ELF is like a plugin, loaded from main firmware. So this ELF module uses absolute addresses from main firmware. I have uploaded better version, it has symbol names for absolute addresses.

**Comment by denys-volodarskyi** [comment]
Ghidra seems load it correctly, can be used as a reference. Also it loads image at 0xa0000000, not sure if it matters.

**Comment by galenbwill** [comment]
Yes, Binary Ninja also wants to load at 0xA0000000 by default (that is the correct load address and entry point, according to the ELF headers).

I tried that as well, but I will try ghidra after the 5.0 release is complete.

Where is this new upload? I don't see it.

**Comment by denys-volodarskyi** [comment]
I've sent it to binaryninja@vector35.com (issue 6215, file2 attachment)
